src/module.c

```diff
@@ -261,6 +264,10 @@ typedef struct RedisModuleBlockedClient {
 static pthread_mutex_t moduleUnblockedClientsMutex = PTHREAD_MUTEX_INITIALIZER;
 static list *moduleUnblockedClients;
 
+#define MODULE_MAX_TEMP_CLIENT_COUNT 1024
+static client *moduleTempClients[MODULE_MAX_TEMP_CLIENT_COUNT];
+static size_t moduleTempClientCount;
```

Comment:
maybe 1024 is too high? (clients consume a lot of memory, and in this case after reaching the peak they'll never be released).
maybe 32 is a better constant?

Reply:
```suggestion
#define MODULE_MAX_TEMP_CLIENT_COUNT 1024
static client *moduleTempClients[MODULE_MAX_TEMP_CLIENT_COUNT];
static size_t moduleTempClientCount = 0;
```
?

Reply:
`client` struct is around 17 kb, so 1024 of them will consume around 17 mb. 
We use 2 `client` objects per blocked client. So, I assumed that we can have ~ 512 blocked clients at a time. 
32 is too few, how about 256(4 mb) or 512(8 mb) ?

Reply:
where did you get the ~512 number from?
since the code will keep working even if we exceed the limit, the compromise we need to make is about
1. efficiency - if we take a number that's too low the optimization will not be sufficient
2. memory - if there's a burst of clients we have no mechanism to later reclaim that memory

i'm not comfortable with very high numbers even as high as 256.
if we wanna go that high, i think we need:
1. some introspection about this memory consumption in INFO and MEMORY commands.
2. maybe a "cron" mechanism to release them after some idle period.

@yossigo WDYT?

Reply:
512 is just an assumption mostly considering redisraft use case. I agree there is trade-off, can consume some memory after reaching the peak. 

Also, there is an opportunity to cut the number half, could you take look at here : 
https://github.com/redis/redis/pull/9940/files/3bbd7a36f7f3191c3c69054e81e58b29890b25dc#r775382159